### ISSUE TITLE
request - option to not log anything 

### ISSUE THREAD

**Issue by redcat42** [aj3423/SpamBlocker#128]
hope all is well, I see you have been busy working on the app

I guess you decided not to publish it on google playstore 

thanks for your work

I came across this app 
https://play.google.com/store/apps/details?id=com.lifesoftwarelab.android.incomingcallcontrol

I was impressed by its clean ui and no logging, it does not even require permissions to read contacts or call logs 
the only drawback is the app does not allow you to 'not wake screen' for silenced calls and it does not hide silenced calls from notifications 

my use case is all this app does but with notification hiding / blocking 

was curious if there was scope for SpamBlocker to allow and not allow permissions based on what the user wants, for example, I don't need logging and I don't need it for SMS blocking, so I don't give SpamBlocker permissions for SMS 

but if there was a way to opt for logging or not, as my dialer app still will show me the log of silenced calls, so I really don't need to use the SpamBlocker db 

just the option to log or not log 
or based on the choices of the user the permission will be asked from the user ... rather than a rewrite 

don't know if what I am saying makes sense, just thought I'll share my thoughts with you

**Comment by aj3423** [comment]
I'll add an option in the history tab, something like "enable db logging", so you can toggle it off.

About the sms permission, these two icons are actually clickable, you can disable the call/sms blocking individually. 
![image](https://github.com/user-attachments/assets/28529601-21dd-4dc5-b537-b654ea9b8fa9)
And in the latest release v2.0, it only asks for permission when you enable a feature, no more popup at first launch.

That app doesn't require contact permission, because the call information already contains [ContactDisplayName](https://developer.android.com/reference/android/telecom/Call.Details#getContactDisplayName()) and [ContactPhotoUri](https://developer.android.com/reference/android/telecom/Call.Details#getContactPhotoUri()). But this app works for both call and sms, there's no such infomation in the sms broadcast. To support that, there would have to be two implementations, which leads to more complex UI and brings more confusions, I decide to not support it.

**Comment by redcat42** [comment]
makes sense. but the user might also wonder why play store is tagging the app as malicious ...

maybe a separate section for SMS and a separate section for calls or separate tab ... but the permissions explanation will still be a problem , I understand 

user will have to trust, SpamBlocker is also asking permission to read all notifications 

I guess as long as the user is given options, the user feels in control, the moment permissions is asked without options the user feels the app is trying something 

that's today's world :)

**Comment by aj3423** [comment]
The "malicious warning" will appear anyway, because there are more sensitive permissions like call/sms log.

I understand this will improve the app's privacy, but it's a trade off between a bit privacy and "clean UI" :) 

I'll think about it later, maybe there is a way to support it without making the UI too complex.

Thanks for the ideas.

**Comment by redcat42** [comment]
just FYI time taken to search for number calling in contacts is a few seconds more than the the taken to check if contact exists using ContactDisplayName

also

if logging option is off, ui need not show the logging tabs 

thanks for your work 👍

**Comment by aj3423** [comment]
> a few seconds

How do you come to this conclusion? How many contacts do you have? Searching within hundreds of contacts should be instant. Do you see a few seconds delay when testing?

> ui need not show the logging tabs

Why do you want to get rid of them, since they are not bothering. I actually plan to add the options within the tabs.

**Comment by redcat42** [comment]
> How do you come to this conclusion? How many contacts do you have? Searching within hundreds of contacts should be instant. Do you see a few seconds delay when testing?

I noticed a few seconds delay when I tested calling myself from another number, the phone making the call rings a few times before the phone receiving the call actually rings 
however, testing in comparison with the other mentioned earlier, the phone rings almost instantly 
I don't mean to compare apps, its just want I noticed and now that you pointed out the different method used by both apps, it seems the method more efficient is using the app as the default spam blocking app in android and using that api

> Why do you want to get rid of them, since they are not bothering. I actually plan to add the options within the tabs.

personally, anything logging is alarming, for example, I've been using version 1.7 for a long time, but I let the SpamBlocker do its thing, I never open it after setting it up, I only use the quicktile to enable/disable when I'm expecting an unknown number to call. I recently opened the SpamBlocker app and it had a huge log, I don't mind the space the log uses but I didn't like seeing it had logged so much, even though I was regularly clearing my phone app call log, in principle any app logging makes me uncomfortable as to why it would do so ... its a personal paranoia I guess, nothing to do with SpamBlocker specifically

**Comment by aj3423** [comment]
> I noticed a few seconds delay when I tested calling myself from another number


Does that delay happen every time? Could it be just a temporary carrier problem? Otherwise it could be a bug. The Testing procedure also contains contact matching, if that returns instantly, then at least the delay is not caused by the contact matching. 

Are you still using the v1.7?  I remember that there was a bug in the old version, the ringing will be delayed for 5 seconds. It has been fixed for long.

About the logs, I see it's necessary to add an option for not logging, but it's unnecessary to hide the tabs at all.

**Comment by redcat42** [comment]
> Are you still using the v1.7? I remember that there was a bug in the old version, the ringing will be delayed for 5 seconds. It has been fixed for long.

I'm still on 1.7, I'll update and check ;) 

thanks for your work

**Comment by aj3423** [comment]
Supported in the action build: https://github.com/aj3423/SpamBlocker/actions/runs/10854854467
Set the "Expire" to 0 to completely disable the logging.
![image](https://github.com/user-attachments/assets/2b05747e-10ef-474f-94fd-bd00b8c5478d)

**Comment by redcat42** [comment]
super!

**Comment by redcat42** [comment]
> That app doesn't require contact permission, because the call information already contains [ContactDisplayName](https://developer.android.com/reference/android/telecom/Call.Details#getContactDisplayName()) and [ContactPhotoUri](https://developer.android.com/reference/android/telecom/Call.Details#getContactPhotoUri()).

can you make a test apk with the above method used to search for 'contact exists' so I can check the difference in response time, currently, even with the latest version, I'm getting a lag between when the phone I'm calling is ringing

**Comment by aj3423** [comment]
> so I can check the difference in response time
No need for a test apk, you can turn off the Contact swich, then the contact will not be checked at all, I guess you will get the same delay. 
Another way it to use the Testing, not sure if this feature was introduced after v1.7, it's the blue Tube button on the bottom right. It simulates a call, and it is processed by the exact procedure as the real call. Try to turn the Contact switch on and test to see if there is any delay.

> this app does not ask for contacts permission at all
I mean it allows contacts by default, but SpamBlocker gives you option to ignore contacts, which makes it more complex.

**Comment by redcat42** [comment]
thanks for all your work brother ❤️

for me, I just need a simple, lightweight solution, wish you all the best, for now, switching to 

https://play.google.com/store/apps/details?id=com.lifesoftwarelab.android.incomingcallcontrol

as its not asking for contacts permission and is verified by play protect on google play store and most important its very light 

❤️ always

**Comment by aj3423** [comment]
Understand. If you need that app to 'not wake screen', you can fire an feature request to that app, since it's just 1 line of code.

**Comment by redcat42** [comment]
I'm back to SpamBlocker :)

apparently, that app has ads, although I'm blocking ads using custom DNS on android it has access to network (when going to permissions, clicking three dots on top right and selecting show all permissions) 

and access to advertising id, 

from what I understand based on the latest privacy about advertising Id it tracks location based on internet connection(non VPN) and location services (if it is on) 

also uses background mobile data for ads 

your app has no network access, feel safer with it :)

**Comment by aj3423** [comment]
Yeah, that's right, no internet access, nothing can ever be leaked, I respect privacy as much as you do.

**Comment by aj3423** [comment]
I'm still curious about the Testing result, with or without Contacts enabled. This delay could be a new bug.

**Comment by redcat42** [comment]
> I'm still curious about the Testing result, with or without Contacts enabled. This delay could be a new bug.

there is a delay with contacts enabled, it is not a bug, its natural, unless SpamBlocker stores contacts in cache/memory, it will always take time, 

using the 'default spam blocker', is android's way of speeding up the process internally,

its just a few rings, its not a big deal ... its important to do things the way they are meant to be done though ... if a number search through contacts is the correct way, then that's the way... if checking if contact exists is the way then that's the way :)

**Comment by aj3423** [comment]
No it's not natural. It's not iterating through the contact list and check them one by one, it's a database lookup done by Android API, which is also used in apps like YetAnotherCallBlocker. Using the attributes directly is the best way, but not the only way. Is it not strange that you are the only one having this issue?

No need to argue about this, it can be easily proven by:
 
Turn off the Contacts switch, and call it(real call, not Testing), is there a delay?
 - if Yes, then it has nothing to do with contacts matching, we can proceed to figure out the real problem.
 - if No, then something wrong with the contacts matching, I'd add some debug info to trace it.

If you don't think it's necessary, we can just ignore it.

**Comment by redcat42** [comment]
> If you don't think it's necessary, we can just ignore it.

its best to just ignore it 🙏
thank you 🙏

**Comment by aj3423** [comment]
I see, no problem.

**Comment by redcat42** [comment]
I just tested, with version 2.0
I have around 3,000+ contacts
there seems to be no lag 👍

this time I tested with SpamBlocker open, the call gets filtered instantly (it shows up in the SpamBlocker log immediately)

earlier, when I tested, my samsung was in deep sleep, I think, like doze mode of greenify app, 

hope that helps

I think it is not a SpamBlocker bug, it is Samsung OneUI OS that causes this
